core/src/main/java/net/adoptopenjdk/icedteaweb/client/controlpanel/ControlPanel.java

```diff
@@ -241,130 +241,69 @@ public void actionPerformed(ActionEvent e) {
 
     /**
      * Add the different settings panels to the GUI.
-     * 
+     *
      * @return A panel with all the components in place.
      */
     private JPanel createMainSettingsPanel() {
-        jvmPanel =  (JVMPanel) createJVMSettingsPanel();
-        SettingsPanel[] panels = new SettingsPanel[] { new SettingsPanel(Translator.R("CPTabAbout"), createAboutPanel()),
-                new SettingsPanel(Translator.R("CPTabCache"), createCacheSettingsPanel()),
-                new SettingsPanel(Translator.R("CPTabCertificate"), createCertificatesSettingsPanel()),
-                // TODO: This is commented out since this is not implemented yet
-                // new SettingsPanel(Translator.R("CPTabClassLoader"), createClassLoaderSettingsPanel()),
-                new SettingsPanel(Translator.R("CPTabDebugging"), createDebugSettingsPanel()),
-                new SettingsPanel(Translator.R("CPTabDesktopIntegration"), createDesktopSettingsPanel()),
-                new SettingsPanel(Translator.R("CPTabJVMSettings"),jvmPanel),
-                new SettingsPanel(Translator.R("CPTabNetwork"), createNetworkSettingsPanel()),
-                // TODO: This is commented out since this is not implemented yet
-                // new SettingsPanel(Translator.R("CPTabRuntimes"), createRuntimesSettingsPanel()),
-                new SettingsPanel(Translator.R("CPTabSecurity"), createSecuritySettingsPanel()),
-                //todo refactor to work with tmp file and apply as asu designed it
-                new SettingsPanel(Translator.R("CPTabPolicy"), createPolicySettingsPanel()),
-                new SettingsPanel(Translator.R("APPEXTSECControlPanelExtendedAppletSecurityTitle"), new UnsignedAppletsTrustingListPanel(PathsAndFiles.APPLET_TRUST_SETTINGS_SYS.getFile(), PathsAndFiles.APPLET_TRUST_SETTINGS_USER.getFile(), this.config))
-        };
-
-        // Add panels.
-        final JPanel settingsPanel = new JPanel(new CardLayout());
-
-        // Calculate largest minimum size we should use.
-        int height = 0;
-        int width = 0;
-        for (SettingsPanel panel : panels) {
-            JPanel p = panel.getPanel();
-            Dimension d = p.getMinimumSize();
-            if (d.height > height) {
-                height = d.height;
-            }
-            if (d.width > width) {
-                width = d.width;
+        final List<ControlPanelProvider> providers = new ArrayList<>();
```

Comment:
You could convert the providers to a `TreeMap<String, ControlPanelProvider>`. The advantages are:
- fast searching for names
- already sorted by name

If sorting by insertion order is more important than sorting by name you could use a `LinkedHashMap`